data_pipeline/surveymonkey/surveymonkey_etl.py

```diff
@@ -56,14 +57,27 @@ def get_bq_json_for_survey_questions_response_json(
     }
 
 
-def get_survey_answer_details(access_token: str, survey_id: str) -> list:
+def get_survey_answers_in_next_page(
+    access_token: str,
+    page_url: str
+) -> dict:
+    headers = get_surveymonkey_api_headers(access_token)
+    return requests.get(page_url, headers=headers).json()
+
+
+def iter_survey_answer_detail_response(access_token: str, survey_id: str) -> Iterable[dict]:
     headers = get_surveymonkey_api_headers(access_token)
     response = requests.get(
```

Comment:
I guess this could now use `get_survey_answers_in_next_page`? (maybe with another name)

Reply:
Sorry I think I did not get the point here?

Reply:
I mean this seem to be doing exactly the same as. Just that you'd be passing in a different `page_url`.

Reply:
i.e.:

```python
response_json = get_survey_answers_in_next_page(
    access_token,
    f'https://api.surveymonkey.com/v3/surveys/{survey_id}/responses/bulk/?per_page=100'
)
```

But then `get_survey_answers_in_next_page` could be renamed to something like `get_survey_answers_page_response`.

Reply:
I see. I have changed the function name to `iter_survey_answers`. I didn’t add `next page` as it is not getting data for only next page. Hope it seem more clear now.

Reply:
Maybe a confusion. You already have the function `get_survey_answers_in_next_page` which you call further down. My point is that you could call it here as well. At that time `get_survey_answers_in_next_page` could do with a different name because it isn't specific to "next page".

Reply:
Actually the function has been created specifically getting data for the `next page`, otherwise we are using survey_id to request the responses. 

```python
def iter_survey_answers(access_token: str, survey_id: str) -> Iterable[dict]:
    headers = get_surveymonkey_api_headers(access_token)
    response = requests.get(
        f'https://api.surveymonkey.com/v3/surveys/{survey_id}/responses/bulk/?per_page=100',
        headers=headers
    )
    response_json = response.json()
    yield from response_json["data"] # THIS PART IS DIRECT RESPONSE
    while "next" in response_json["links"]: # IF THERE IS NEXT PAGE LINK, THEN WE ARE USING THIS FUNCTION
        response_json = get_survey_answers_in_next_page(access_token, response_json["links"]["next"])
        yield from response_json["data"]
```

Reply:
Maybe a quick call? I think we are talking past each other :-)